src/src/Helpers/Modbus_device.cpp

```diff
@@ -114,104 +140,74 @@ void ModbusDEVICE_struct::processCommand() {
   if (_modbus_link != nullptr) {
     _modbus_link->processCommand(); // Trigger processing of the command queue on the link
   }
-  else {
-    _state = ModbusQueueState_t::ERROR;
-  }
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // Callback function called by the Modbus link when a response is received for a queued request.
 // Note that the response might be an invalid response or a timeout
 // The queueID identifies the request.
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-void ModbusDEVICE_struct::linkCallback(uint16_t queueID)
+void ModbusDEVICE_struct::linkCallback(Modbus_RequestQueueElement *req)
 {
-  if (queueID != _queueID) {
-    return; // Not for us
-  }
-  bool response = _modbus_link->getResponse(_queueID, _recv_buf, _recv_buf_used);
-
-  if (response) {
-    switch (_messageType) {
-      case ModbusMessageType::READ_HOLDING_REGISTERS:
-      {
-        if ((_recv_buf[0] == _modbus_address) && (_recv_buf[1] == MODBUS_READ_HOLDING_REGISTERS) && (_recv_buf[2] == 2)) {
-          uint16_t crc = CalculateCRC(_recv_buf, 5);
-
-          if ((_recv_buf[5] == lowByte(crc)) && (_recv_buf[6] == highByte(crc))) {
-            // Valid response
-            if (_resultPtr != nullptr) {
-              *_resultPtr = (_recv_buf[3] << 8) | _recv_buf[4]; // Combine high and low byte
-            }
-            _state = ModbusQueueState_t::AVAILABLE;
-          } else {
-            // Invalid CRC
-            _state = ModbusQueueState_t::ERROR;
+  String log = F("---> Device callback: ");
+
+  log += req->_id;
+  log += F(", Message = ");
+  log += (uint8_t)req->_messageType;
```

Comment:
Please only use those C-style casts only when needed. (e.g. when literally casting a pointer to an int to get the address is a good example)
In C++ we should use `static_cast<uint8_t>(....)`